src/tests/ftest/util/server_utils.py

```diff
@@ -50,292 +47,352 @@ class ServerFailed(Exception):
+        return True
 
-    Args:
-        hosts (list): list of host names where servers are running
-    """
-    kill_cmds = [
-        "pkill '(daos_server|daos_io_server)' --signal INT",
-        "sleep 5",
-        "pkill '(daos_server|daos_io_server)' --signal KILL",
-    ]
-    # Intentionally ignoring the exit status of the command
-    pcmd(hosts, "; ".join(kill_cmds), False, None, None)
+    def stop(self):
+        """Stop the server through the runner."""
+        if self.runner.job.yaml.is_nvme():
+            storage_reset(self._hosts)
+
+        print("Stopping servers")
+        error_list = []
+        try:
+            self.runner.stop()
+        except CommandFailure as error:
+            error_list.append("Error stopping servers: {}".format(error))
+
+        return error_list
+
+    def prep(self):
+        """Prepare the hosts before starting daos server.
+
+        Args:
+            hosts (list): list of host names to prepare
+
+        Raises:
+            ServerFailed: if there is any errors preparing the hosts
+
+        """
+        # Kill any doas servers running on the hosts
+        self.kill()
+
+        # Clean up any files that exist on the hosts
+        self.clean(scm=True)
+
+        # Verify the domain socket directory is present and owned by this user
+        user = getpass.getuser()
+        file_checks = (
+            ("Server", self._hosts, "/var/run/daos_server"),
+        )
+        for host_type, host_list, directory in file_checks:
+            status, nodeset = check_file_exists(host_list, directory, user)
+            if not status:
+                raise ServerFailed(
+                    "{}: {} missing directory {} for user {}.".format(
+                        nodeset, host_type, directory, user))
+
+    def kill(self):
+        """Forcably kill any daos server processes running on hosts.
+
+        Sometimes stop doesn't get everything.  Really whack everything
+        with this.
+
+        """
+        kill_cmds = [
+            "pkill '(daos_server|daos_io_server)' --signal INT",
+            "sleep 5",
+            "pkill '(daos_server|daos_io_server)' --signal KILL",
+        ]
+        # Intentionally ignoring the exit status of the command
+        result = pcmd(self._hosts, "; ".join(kill_cmds), False, None, None)
+        if len(result) > 1 or 0 not in result:
+            raise ServerFailed(
+                "Error cleaning tmpfs on servers: {}".format(
+                    ", ".join(
+                        [str(result[key]) for key in result if key != 0])))
+
+    def clean(self, scm=False):
+        """Clean the tmpfs on the servers.
+
+        Args:
+            scm (bool): if true, remove tmpfs mount
+        """
+        clean_cmds = [
+            "find /mnt/daos -mindepth 1 -maxdepth 1 -print0 | xargs -0r rm -rf"
+        ]
+
+        if self.runner.job.yaml.is_scm() and scm:
+            clean_cmds.append("sudo umount /mnt/daos; sudo rm -rf /mnt/daos")
```

Comment:
Why are you cleaning the mount point (`/mnt/daos` and it's contents) under the SCM?

Reply:
The mount needs to be cleanedup in order to trigger the "format" mode when starting the server.

Reply:
I'm not sure I understand.  Let me try to rephrase...

Without considering a tmpfs or SCM for the moment, let's just acknowledge that `/mnt/daos` is just a directory on the root filesystem of a Linux machine.  It can be empty or not.

In order to use DAOS, we need to give it some storage to use.  We do that by mounting a storage device on `/mnt/daos`.  If there were files in the `/mnt/daos` directory on that root filesystem they are now hidden underneath whatever we mounted on `/mnt/daos` and now we see the contents of that storage device (tmpfs or SCM) at `/mnt/daos`.

In the above code, you are _unmounting_ whatever storage was mounted on `/mnt/daos` and then removing any files that were in `/mnt/daos` **on the root filesystem**.  That's bad.  But moreover, I'm failing to see how removing the contents of the `/mnt/daos` directory on the root filesystem (recall, the storage that was mounted at `/mnt/daos` has now been unmounted) triggers a format when starting the server.  Surely when we want to format we will mount the storage device on `/mnt/daos` before we try to format it, thereby hiding the files that are on `/mnt/daos` yes?

Reply:
@brianjmurrell @amjustin13 Would flipping the order resolve this?
```suggestion
            clean_cmds.append("sudo rm -rf /mnt/daos; sudo umount /mnt/daos")
```